langchain-core/src/runnables/base.ts

```diff
@@ -378,7 +382,17 @@ export abstract class Runnable<
     delete config.runId;
     let output;
     try {
-      output = await func.call(this, input, config, runManager);
+      const promise = func.call(this, input, config, runManager);
+      output = options?.signal
+        ? await Promise.race([
+            promise,
+            new Promise<never>((_, reject) => {
```

Comment:
Ah, edge case but what if someone passes an already aborted signal/there's some race condition?

Feel like there should be a check?